pkg/cmd/server/origin/auth.go

```diff
@@ -375,8 +388,17 @@ func (c *AuthConfig) getAuthenticationHandler(mux cmdutil.Mux, errorHandler hand
 				}
 				passwordSuccessHandler := handlers.AuthenticationSuccessHandlers{c.SessionAuth, redirectSuccessHandler{}}
 
+				loginPath := openShiftLoginPrefix
+
+				if multiplePasswordProviders {
+					// If there is more than one Identity Provider acting as a login
+					// provider, we need to give each of them their own login path,
+					// to avoid ambiguity.
+					loginPath = path.Join(openShiftLoginPrefix, identityProvider.Name)
```

Comment:
I *think* we need to URL-encode the name here, but double-check with an IDP name like "baz qux"